arelle/plugin/validate/NL/rules/fr_nl.py

```diff
@@ -17,11 +15,40 @@
     DISCLOSURE_SYSTEM_NT17,
     DISCLOSURE_SYSTEM_NT18,
 )
-from ..PluginValidationDataExtension import PluginValidationDataExtension
+from ..PluginValidationDataExtension import PluginValidationDataExtension, BYTE_ORDER_MARK_BYTES, CDATA_END_SEQUENCE_PATTERN
 
 _: TypeGetText
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+        DISCLOSURE_SYSTEM_NT18,
+    ],
+)
+def rule_fr_nl_1_01(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FR-NL-1.01: A BOM character MUST NOT be used.
+    """
+    for basename, lineNumber, match in pluginData.findDisallowedPatterns(val.modelXbrl):
+        matchEncoded = match.encode()
```

Comment:
Perhaps `findDisallowedPatterns` should do the `encode()`?

I'm somewhat confused how this all works.  It seems like we're reading as bytes, running a regex on bytes, so the match results are in bytes, so we need to encode here in order to compare against the str literal, is that right?  If so, why isn't `encode()` needed in `rule_fr_nl_2_06`?

Reply:
Per Steven's comment, ended up not going with the approach of checking for BOM throughout the file. It seems BOM is, by definition, a character at the beginning of a text stream. So the whole regex/pattern matching part of this is abandoned in favor of simply checking the first character.